makefiles/dependency_resolution.inc.mk

```diff
@@ -50,4 +50,13 @@ else
     $(shell $(COLOR_ECHO) "$(COLOR_RED)Deprecated modules are in use:$(COLOR_RESET)"\
                           "$(DEPRECATED_MODULES_USED)" 1>&2)
   endif
+  # Warn about used deprecated boards
+  include $(RIOTMAKE)/deprecated_boards.inc.mk
+  DEPRECATED_BOARDS_USED := $(sort $(filter $(DEPRECATED_BOARDS),$(BOARD)))
```

Comment:
Wait actually can you use `memoized` here?

Reply:
Now that I wrapped it with `ifeq ($(MAKELEVEL),0)` does it make sense.  I also removed the sort.  The problem space is maybe even smaller since we won't have a list of boards...

If only I could do:
```py
if BOARD in DEPRECATED_BOARDS:
  raise RIOT_EXCEPTION(f'{BOARD} is deprecated')

Reply:
True.